platform/jvm/Shell.kt

```diff
@@ -0,0 +1,74 @@
+package com.vaticle.bazel.distribution.platform.jvm
+
+import com.vaticle.bazel.distribution.platform.jvm.JVMPlatformAssembler.logger
+import com.vaticle.bazel.distribution.platform.jvm.Shell.Command.Companion.arg
+import org.zeroturnaround.exec.ProcessExecutor
+import org.zeroturnaround.exec.ProcessResult
+import java.nio.file.Path
+import java.nio.file.Paths
+
+class Shell(private val verbose: Boolean = false, private val printSensitiveData: Boolean = false) {
```

Comment:
Depending on the logging settings `verbose` and `printSensitiveData`, `Shell` will behave differently in terms of how it outputs the commands being entered, and how it outputs the _output_ of the commands being run.

The Shell class is set up with a security-first mentality; providing the transparent ability to pass arguments and explicitly mark them as non-printable. This is useful for masking passwords.